vendor/github.com/go-check/check/run.go

```diff
@@ -23,28 +25,48 @@ func Suite(suite interface{}) interface{} {
 }
 
 // -----------------------------------------------------------------------
+// strslice implements flag.Value
+
+type strslice []string
+
+func (ss *strslice) String() string {
+	return fmt.Sprintf("%v", []string(*ss))
+}
+
+func (ss *strslice) Set(value string) error {
+	*ss = append(*ss, value)
+	return nil
+}
+
+// -----------------------------------------------------------------------
 // Public running interface.
 
 var (
-	oldFilterFlag  = flag.String("gocheck.f", "", "Regular expression selecting which tests and/or suites to run")
+	oldFilterFlag  strslice
 	oldVerboseFlag = flag.Bool("gocheck.v", false, "Verbose mode")
 	oldStreamFlag  = flag.Bool("gocheck.vv", false, "Super verbose mode (disables output caching)")
 	oldBenchFlag   = flag.Bool("gocheck.b", false, "Run benchmarks")
 	oldBenchTime   = flag.Duration("gocheck.btime", 1*time.Second, "approximate run time for each benchmark")
 	oldListFlag    = flag.Bool("gocheck.list", false, "List the names of all tests that will be run")
 	oldWorkFlag    = flag.Bool("gocheck.work", false, "Display and do not remove the test working directory")
 
-	newFilterFlag  = flag.String("check.f", "", "Regular expression selecting which tests and/or suites to run")
-	newVerboseFlag = flag.Bool("check.v", false, "Verbose mode")
-	newStreamFlag  = flag.Bool("check.vv", false, "Super verbose mode (disables output caching)")
-	newBenchFlag   = flag.Bool("check.b", false, "Run benchmarks")
-	newBenchTime   = flag.Duration("check.btime", 1*time.Second, "approximate run time for each benchmark")
-	newBenchMem    = flag.Bool("check.bmem", false, "Report memory benchmarks")
-	newListFlag    = flag.Bool("check.list", false, "List the names of all tests that will be run")
-	newWorkFlag    = flag.Bool("check.work", false, "Display and do not remove the test working directory")
-	checkTimeout   = flag.String("check.timeout", "", "Panic if test runs longer than specified duration")
+	newFilterFlag    strslice
+	newVerboseFlag   = flag.Bool("check.v", false, "Verbose mode")
+	newStreamFlag    = flag.Bool("check.vv", false, "Super verbose mode (disables output caching)")
+	newBenchFlag     = flag.Bool("check.b", false, "Run benchmarks")
+	newBenchTime     = flag.Duration("check.btime", 1*time.Second, "approximate run time for each benchmark")
+	newBenchMem      = flag.Bool("check.bmem", false, "Report memory benchmarks")
+	newListFlag      = flag.Bool("check.list", false, "List the names of all tests that will be run")
+	newWorkFlag      = flag.Bool("check.work", false, "Display and do not remove the test working directory")
+	checkTimeout     = flag.String("check.timeout", "", "Panic if test runs longer than specified duration")
+	checkFiltersFile = flag.String("check.ffile", "", "A path to a LF-separated file that contains filter strings (OR-match)")
```

Comment:
`-check.ffile` is not needed and removed in the PR for upstream: https://github.com/cpuguy83/check/pull/4